src/github/client/github-client.ts

```diff
@@ -26,5 +30,42 @@ export class GitHubClient {
 			this.restApiUrl = `${baseUrl}/api/v3`;
 			this.graphqlUrl = `${baseUrl}/api/graphql`;
 		}
+
+		this.axios = axios.create({
+			baseURL: this.restApiUrl,
+			transitional: {
+				clarifyTimeoutError: true
+			},
+			... this.getProxyConfig(this.restApiUrl)
+		});
 	}
+
+	public getProxyConfig = (baseUrl: string): Partial<AxiosRequestConfig> => {
```

Comment:
/nit
Happy to leave as it is, just minor suggestion.

-------------

I know it is tricky to test, and therefore change from `private` to `public`.

In this case, I'd usually do bellow in the test file, so instead of testing private method, assert on the dependencies. 
```javascript
import axios from "axios";
it('should ...', async ()=>{
   jest.spyOn(axios, 'create');
   ... do stuff
   expect(axios.get).toBeCalledWith(expect.objectContaining({
          proxy: { agent: blah blah } 
   });
});
```

Another option will be using nock to assert the http call is going through proxy I guess, but seems much more code.

Reply:
It doesn't work with Nock, I had researched that. 

I had thought about mocking the axios client, but the current test is so much easier to read and maintain so I opted for that (even though it doesn't test the ultimate axios call).